### PR TITLE
Add Guide to NumPY

### PR DESCRIPTION
**Pull request by hstsethi**
## What does this PR do?
Add resource 

### Description

While this book may be bit dated, I found it to be extremely helpful. It has lots of explanations, and almost all code samples still work.

### Why is this valuable (or not)?

This book was authored by a professor at MIT. 

### How do we know it's really free?
The license says so. It reads:

> After October 31, 2010 this book may be freely copied in any format and used as source material for other books as long as ac-knowledgement of the original author is given.

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [X] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction).
- [X] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
There's another numpy book in the python main section; if you want to start a new section move that one too, so you don't have an orphan.

**Comment by hstsethi** [comment]
> There's another numpy book in the python main section; if you want to start a new section move that one too, so you don't have an orphan.

I am sorry for this again but I'm not sure which Numpy book you are referring to.  I have checked both Indices: by subject, by language, but I couldn't find a single Numpy book. There are two SciPY books which have a chapter on NumPY but they aren't exclusive for Numpy.

And yes, I do believe so that Numpy should have a separate section, like Pandas, as it's used for variety of tasks, not just machine learning. But if you want, I can merge it into Python section too.

**Comment by eshellman** [comment]
"From Python to NumPy"

**Comment by hstsethi** [comment]
@eshellman I have done as you said so. Please review.

I am sorry for inconvenience caused. I was on mobile web which doesn't have a good search functionality.

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -2128,6 +2128,14 @@ Books on general-purpose programming that don't focus on a specific language are
 * [Kivy Programming Guide](https://kivy.org/docs/guide-index.html)
 
 
+#### NumPY
+
+* [From Python to NumPy](https://www.labri.fr/perso/nrougier/from-python-to-numpy/) - Nicolas P. Rougier (HTML) (3.6)
+* [Guide to NumPY 2010](https://web.mit.edu/dvp/Public/numpybook.pdf) - Travis E. Oliphant (PDF).
+* [NumPy: the absolute basics for beginners](https://numpy.org/doc/stable/user/absolute_beginners.html) - NumPy developers (HTML).
```
Comment (eshellman) [inline books/free-programming-books-langs.md:2135]: This is a chapter in the NumPy user guide, so it's redundant.

**Pull request merged**